wafer/users/views.py

```diff
@@ -38,6 +38,13 @@ class ProfileView(DetailView):
     slug_field = 'username'
     slug_url_kwarg = 'username'
 
+    def get_object(self, *args, **kwargs):
+        object_ = super(ProfileView, self).get_object(*args, **kwargs)
+        if not settings.WAFER_PUBLIC_ATTENDEE_LIST:
+            if not object_.userprofile.accepted_talks().exists():
+                raise Http404()
```

Comment:
Surely this is too restrictive? Logged in users should be able to see their own profile, even if they don't have an accepted talk.